include/wil/resource.h

```diff
@@ -27,7326 +27,7575 @@
+{
+    static_assert(
+        wistd::is_same<typename unique_any_t<policy>::policy::pointer_invalid, wistd::nullptr_t>::value,
+        "the resource class does not use nullptr as an invalid value");
+    return !!right;
+}
 
-        protected:
-            TLambda m_lambda;
-            bool m_call = true;
-        };
+template <typename policy>
+bool operator<(const unique_any_t<policy>& left, const unique_any_t<policy>& right) WI_NOEXCEPT
+{
+    return (left.get() < right.get());
+}
 
-#ifdef WIL_ENABLE_EXCEPTIONS
-        template <typename TLambda>
-        class lambda_call_log
-        {
-        public:
-            lambda_call_log(const lambda_call_log&) = delete;
-            lambda_call_log& operator=(const lambda_call_log&) = delete;
-            lambda_call_log& operator=(lambda_call_log&& other) = delete;
+template <typename policy>
+bool operator>=(const unique_any_t<policy>& left, const unique_any_t<policy>& right) WI_NOEXCEPT
+{
+    return (!(left < right));
+}
 
-            explicit lambda_call_log(void* address, const DiagnosticsInfo& info, TLambda&& lambda) WI_NOEXCEPT :
-            m_address(address), m_info(info), m_lambda(wistd::move(lambda))
-            {
-                static_assert(wistd::is_same<decltype(lambda()), void>::value, "scope_exit lambdas must return 'void'");
-                static_assert(!wistd::is_lvalue_reference<TLambda>::value && !wistd::is_rvalue_reference<TLambda>::value,
-                    "scope_exit should only be directly used with a lambda");
-            }
+template <typename policy>
+bool operator>(const unique_any_t<policy>& left, const unique_any_t<policy>& right) WI_NOEXCEPT
+{
+    return (right < left);
+}
 
-            lambda_call_log(lambda_call_log&& other) WI_NOEXCEPT :
-            m_address(other.m_address), m_info(other.m_info), m_lambda(wistd::move(other.m_lambda)), m_call(other.m_call)
-            {
-                other.m_call = false;
-            }
+template <typename policy>
+bool operator<=(const unique_any_t<policy>& left, const unique_any_t<policy>& right) WI_NOEXCEPT
+{
+    return (!(right < left));
+}
 
-            ~lambda_call_log() WI_NOEXCEPT
-            {
-                reset();
-            }
+// unique_any provides a template alias for easily building a unique_any_t from a unique_storage class with the given
+// template parameters for resource_policy.
+
+template <
+    typename pointer,                                      // The handle type
+    typename close_fn_t,                                   // The handle close function type
+    close_fn_t close_fn,                                   //      * and function pointer
+    typename pointer_access = details::pointer_access_all, // all, noaddress or none to control pointer method access
+    typename pointer_storage = pointer,          // The type used to store the handle (usually the same as the handle itself)

```

Comment:
Despite `AlignTrailingComments: true`, I can't seem to figure out a way to keep the comments aligned. Not urgent or pressing; I just feel like it should work